### DISCUSSION TITLE
Choice of forecast pressure level to interpolate to level of observation.

### DISCUSSION THREAD

**Discussion by jwraby** [dtcenter/METplus#2459] [Configuration] [answered]
Using Point-Stat V11.1.0, my config file has the forecast and observation field setting of P862-876. The WRF output was post-processed to output on pressure levels 862, 869, and 876. The OBS_LVL of my ADPUPA observations are 868.3243 (TMP) and 869.2578 for UGRD and VGRD. How does Point-Stat choose the forecast output pressure level to interpolate vertically to the OBS_LVL of the observation? Thanks.

**Comment by CPKalb** [comment, marked as answer]
Hi:

For vertical interpolation, if the forecast and observations are at the same level, no interpolation is performed.  If they are on different levels, vertical interpolation is performed in the natural log of pressure coordinates (for all variables except specific humidity).  In your case, the levels are not exact due to the decimal points, so they will be interpolated.   Specifically, the forecasts will be interpolated to the observation level. 

Which of the three forecast levels MET will use to do the interpolation to the observation level depends on how your configuration is set up for the forecast variable.  If you want it to use the closest, you would want to specify P869 as the level for your forecast variables.

Christina

**Reply by jwraby** [reply]
Christina - 
Thanks for your reply. Good suggestion on the level to use for the closest forecast. I had just completed a test where I used P869 for the forecast field and P862-876 for the observations and I ended up with the desired matching of the available observations with the forecast on the 869 pressure level interpolated vertically to the level of the observation. Great minds! When I did the earlier test, I used P862-876 for both forecast and observations fields and the forecast value which was matched with the same TMP observation was different. In that earlier case (when there were multiple forecast levels specified) do you know how it chose the forecast value for interpolation/matching? Just trying to gain more understanding of how Point-Stat does vertical level verification.

**Reply by jwraby** [reply]
Additional info. I have the Point-Stat log files for verbosity=5 for these runs and I was unable to immediately see information which might answer these questions. Perhaps I missed something there? Thanks.

**Reply by CPKalb** [reply]
My best guess would be that when you have the range specified, there would be a warning stating that Point-Stat found multiple matches for the forecast field and so used the first one it found (this is how I have seen Point-Stat behave although I have not directly tested using a range of levels).  I would need to see the log file to verify that this is actually what happened.  Could you post the log file to this discussion?

If that information isn't in the log file, I can check with some of our software engineers over here to see how it is specifically handled.

Christina

**Reply by jwraby** [reply]
I will try to add the log file.

**Reply by CPKalb** [reply]
You will likely have to give it a .txt extension to add the file.

**Reply by jwraby** [reply]
I tried several times to upload the log file, but I can't get it to work.

**Reply by jwraby** [reply]
I added .txt to the file and tried again, but no change.

**Reply by CPKalb** [reply]
The other option is to upload it to our ftp site using the [instructions](https://github.com/dtcenter/METplus/discussions/954).

Christina

**Reply by jwraby** [reply]
Thanks for the instructions. I can't seem to make it work. I think my question is answered already, so I'll end this discussion. Appreciate the information which has helped me.
